packages/perseus/src/widgets/interactive-graphs/graphs/circle.tsx

```diff
@@ -38,3 +38,91 @@ export function CircleGraph(props: CircleGraphProps) {
         </>
     );
 }
+
+function MovableCircle(props: {
+    center: vec.Vector2;
+    radius: number;
+    onMove: (newCenter: vec.Vector2) => unknown;
+}) {
+    const {center, radius, onMove} = props;
+    const {snapStep} = useGraphConfig();
+
+    const draggableRef = useRef<SVGGElement>(null);
+
+    const {dragging} = useMovable({
+        gestureTarget: draggableRef,
+        point: center,
+        onMove,
+        constrain: (p) => snap(snapStep, p),
+    });
+
+    const [centerPx] = useTransformVectorsToPixels(center);
+    const [radiiPx] = useTransformDimensionsToPixels([radius, radius]);
+
+    return (
+        <g
+            ref={draggableRef}
+            tabIndex={0}
+            className={`movable-circle ${dragging ? "movable-circle--dragging" : ""}`}
+        >
+            {/* focus ring */}
+            <ellipse
+                className="focus-ring"
+                cx={centerPx[0]}
+                cy={centerPx[1]}
+                rx={radiiPx[0] + 3}
+                ry={radiiPx[1] + 3}
+            />
+            <Circle
+                center={center}
+                radius={radius}
+                fillOpacity={0}
+                color={color.blue}
+            />
+            <DragHandle center={center} />
+        </g>
+    );
+}
+
+const dragHandleDimensions: vec.Vector2 = [24, 14];
+const dragHandlePointPositions = crossProduct([-4.4, 0, 4.4], [-2.1, 2.1]);
```

Comment:
Just curious - where did these numbers come from?

Reply:
I fiddled with them until the drag handle looked close to the design mock and reasonably legible.